### PR TITLE
ext_auth support backendRefs

### PR DESCRIPTION
**Pull request by zirain**
xref: https://github.com/envoyproxy/gateway/issues/3091

### CODE REVIEW

File: internal/provider/kubernetes/indexers.go [hunk 1]
```diff
@@ -412,9 +412,17 @@ func backendSecurityPolicyIndexFunc(rawObj client.Object) []string {
 
 	if securityPolicy.Spec.ExtAuth != nil {
 		if securityPolicy.Spec.ExtAuth.HTTP != nil {
-			backendRef = &securityPolicy.Spec.ExtAuth.HTTP.BackendRef
+			http := securityPolicy.Spec.ExtAuth.HTTP
+			backendRef = http.BackendRef
+			if len(http.BackendRefs) > 0 {
+				backendRef = v1alpha1.ToBackendObjectReference(http.BackendRefs[0])
```
Comment (arkodg) [inline internal/provider/kubernetes/indexers.go:418]: looks like we are skipping indexing backendRefs[1]...... so if they change we won't be able to reconcile

File: internal/gatewayapi/securitypolicy.go [hunk 2]
```diff
@@ -775,16 +772,24 @@ func (t *Translator) buildExtAuth(
...
 	// should have caught them
-	case http == nil && grpc == nil:
-		return nil, errors.New("one of grpc or http must be specified")
 	case http != nil && grpc != nil:
 		return nil, errors.New("only one of grpc or http can be specified")
 	case http != nil:
-		backendRef = &http.BackendRef
+		backendRef = http.BackendRef
+		if len(http.BackendRefs) != 0 {
+			backendRef = egv1a1.ToBackendObjectReference(http.BackendRefs[0])
```
Comment (arkodg) [inline internal/gatewayapi/securitypolicy.go:780]: will the support for multiple backendRefs be in a follow up ?

**Review by arkodg — DISMISSED** [review]
LGTM thanks !

**Comment by arkodg** [comment]
just realized, this is the Api PR

**Comment by zirain** [comment]
> just realized, this is the Api PR :)

everything is here.

> will the support for multiple backendRefs be in a follow up ?

is there any user case for this?

**Comment by zirain** [comment]
wait https://github.com/envoyproxy/gateway/pull/3478

**Review by arkodg — APPROVED** [review]

File: api/v1alpha1/ext_auth_types.go [hunk 3]
```diff
@@ -55,19 +55,42 @@ type ExtAuth struct {
...
 	// BackendRef references a Kubernetes object that represents the
 	// backend server to which the authorization request will be sent.
 	// Only service Kind is supported for now.
-	BackendRef gwapiv1.BackendObjectReference `json:"backendRef"`
+	// Deprecated: Use BackendRefs instead.
+	BackendRef *gwapiv1.BackendObjectReference `json:"backendRef,omitempty"`
+
+	// BackendRefs references a Kubernetes object that represents the
+	// backend server to which the authorization request will be sent.
+	// Only service Kind is supported for now.
```
Comment (shawnh2) [inline api/v1alpha1/ext_auth_types.go:68]:
```suggestion
	// Only Service kind is supported for now.
```

**Review by shawnh2 — APPROVED** [review]
LGTM! Just some nits.

**Pull request merged**